workspaces/mcp-integrations/plugins/backstage-mcp-plugin-backend/src/plugin.ts

```diff
@@ -41,33 +41,123 @@ export const backstageMcpPlugin = createBackendPlugin({
-          schema: {
-            input: z =>
-              z.object({
-                name: z.string().describe('The name of the person to greet'),
-              }),
-            output: z =>
-              z.object({
-                greeting: z.string().describe('The generated greeting'),
-              }),
-          },
-          action: async ({ input }) => ({
-            output: { greeting: `Hello ${input.name}!` },
-          }),
-        });
         // This action is used to fetch the list of catalog entities from Backstage. It returns an array of entity names
         actionsRegistry.register({
           name: 'fetch-catalog-entities',
           title: 'Fetch Catalog Entities',
-          description:
-            'Retrieve the list of catalog entities from the Backstage server.',
+          description: `Search and retrieve catalog entities from the Backstage server.
+
+List all Backstage entities such as Components, Systems, Resources, APIs, Locations, Users, and Groups. 
+Results are returned in JSON array format, where each entry in the JSON array has the following fields: 'name', 'description','uid', and 'type'.
+
+This tool searches through the software catalog to find components the entities. It supports filtering by entity properties and 
+text-based search across entity metadata.
+
+Examples:
+  # Get all entities in the catalog
+  fetch-catalog-entities
+  Output: {
+  "entities": [
+    {
+      "name": "model-service-api",
+      "kind": "API",
+      "tags": [
+        "api",
+        "openai",
+        "vllm"
+      ]
+    },
+    {
+      "name": "developer-model-service",
+      "kind": "Component",
+      "tags": [
+        "genai",
+        "ibm-granite",
+        "vllm",
+        "llm",
+        "developer-model-service",
+        "authenticated",
+        "gateway"
+      ]
+    },
+    {
+      "name": "generated-c4d4657b4fbb886fe0a962cdf12b8732d33946ca",
+      "kind": "Location",
+      "tags": []
+    },
+    {
+      "name": "ibm-granite-8b-code-instruct",
+      "kind": "Resource",
+      "tags": [
+        "genai",
+        "ibm",
+        "llm",
+        "granite",
+        "conversational",
+        "task-text-generation"
+      ]
+    }
+  ]
+}
+
+  # Find all entities of kind Resource
+  fetch-catalog-entities kind:Resource
+  Output: {
+  "entities": [
+    {
+      "name": "ibm-granite-8b-code-instruct",
+      "kind": "Resource",
+      "tags": [
+        "genai",
+        "ibm",
+        "llm",
+        "granite",
+        "conversational",
+        "task-text-generation"
+      ]
+    }
+  ]
+}
+
+  # Find all Components of type service
+  fetch-catalog-entities kind:Resource type:storage
+  Output: {
+  "entities": [
+    {
+      "name": "ibm-granite-s3-bucket",
+      "kind": "Resource",
+      "type": "storage",
+      "tags": [
+        "genai",
+        "ibm",
+        "llm",
+        "granite",
+        "conversational",
+        "task-text-generation"
+      ]
+    }
+  ]
+}
+`,
```

Comment:
and "end of description" comment might be helpful ... it took me a couple to sort out that `schema` was the next field in the parameter passed to the actions registry